ale_linters/erlang/erlc.vim

```diff
@@ -3,7 +3,8 @@
 let g:ale_erlang_erlc_options = get(g:, 'ale_erlang_erlc_options', '')
 
 function! ale_linters#erlang#erlc#GetCommand(buffer) abort
-    return 'erlc ' . g:ale_erlang_erlc_options . ' %t'
+    let l:temp_dir = has('win32') ? $TMP : $TMPDIR
```

Comment:
I recommend using the Vim `tempname()` function instead. Files created with `tempname()` will be automatically deleted when Vim is closed.

There is also a function used in `ale/engine.vim` which automatically deletes files as best as it can when a lint cycle ends, as well as another one for generating a temporary filename using `tempname()` securely. That second one might need to be moved to `ale/util.vim`.